app/src/main/java/ar/com/wolox/android/example/ui/home/news/NewsPresenter.kt

```diff
@@ -0,0 +1,86 @@
+package ar.com.wolox.android.example.ui.home.news
+
+import ar.com.wolox.android.example.model.New
+import ar.com.wolox.android.example.network.NewsService
+import ar.com.wolox.wolmo.core.presenter.BasePresenter
+import ar.com.wolox.wolmo.networking.retrofit.RetrofitServices
+import okhttp3.Headers
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
+import java.net.UnknownHostException
+import javax.inject.Inject
+
+class NewsPresenter @Inject constructor(private val mRetrofitServices: RetrofitServices) : BasePresenter<INewsView>() {
+
+    var nextPage: String? = null
+
+    fun hasMore(): Boolean = nextPage != null
+
+    fun requestNews(paginated: Boolean) {
+        val newsService = mRetrofitServices.getService(NewsService::class.java)
+
+        if (paginated && nextPage == null) {
+            runIfViewAttached(INewsView::hideRefreshing)
+            return
+        }
+
```

Comment:
I think you must check previously if `nextPage` is null.

Reply:
Last `if` condition does that, did you mean something else?

Reply:
Oh, okey. Now I get what you did. Yes, it's okey how it's done. Forget what I said, I'm sorry. 😅